js/components/grid/calendar.js.jsx

```diff
@@ -120,23 +120,17 @@ class TimetableBody extends React.Component {
   render() {
     let rows = [];
     // For each row from 8 o'clock to 22 o'clock, there is an 'Hour' and 'Half hour' row
-    for (let i = 8; i < 22; i++) {
+    for (let i = 8; i < 22; i+=0.5) {
       rows.push(
         <TimetableRow
           session={this.props.session}
           time={i}
           key={'timetable-row-' + i + this.props.session}
           currentLectures={this.props.lectures[i]}
-          previousLectures={this.props.lectures[i-1]}
+          previousLectures={this.props.lectures[i-0.5]}
```

Comment:
This type of change isn't good; you shouldn't try to index into a list using a fractional value. You need to come up with a formula that maps position in this array (or other arrays) with time of day.

Reply:
Hmmm this is okay now then, but still not ideal.